src/core/adapters/personalization.ts

```diff
@@ -0,0 +1,80 @@
+import Actions from '../actions';
+import Configuration from '../configuration';
+import Selectors from '../selectors';
+import Store from '../store';
+
+namespace Personalization {
+  export const extractBias = ({ payload }: Actions.SelectRefinement, store: Store.State) => {
+    const config = Selectors.config(store).personalization.realTimeBiasing;
+    const byId = Selectors.realTimeBiasesById(store);
+    const { value, field } = Selectors.refinementCrumb(store, payload.navigationId, payload.index);
+
+    if (!config.attributes[field]) {
+      return null;
+    }
+
+    return {
+      variant: field,
+      key: value,
+      bias: (byId[field] && byId[field][value]) ? {
+        ...byId[field][value],
+        lastUsed: Math.floor(Date.now() / 1000)
+      } : generateNewBias()
+    };
+  };
+
+  export const generateNewBias = () => ({
+    lastUsed: Math.floor(Date.now() / 1000)
+  });
+
+  // tslint:disable-next-line max-line-length
+  export const transformToBrowser = (state: Store.Personalization.Biasing, reducerKey: string): BrowserStorageState => ({
+    expiry: state.expiry,
+    allIds: state.allIds ? state.allIds.map(({ variant, key }) => ({
+      variant,
+      key,
+      ...state.byId[variant][key]
+    })) : []
+  });
+
+  // tslint:disable-next-line max-line-length
+  export const transformFromBrowser = (state: BrowserStorageState, reducerKey: string): Store.Personalization.Biasing => {
+    const olderThanTime = Math.floor(Date.now() / 1000) - state.expiry;
+    const filteredState = state.allIds.filter((element) => element.lastUsed >= olderThanTime);
+    let allIds = [];
+    let byId = {};
+    filteredState.forEach(({ variant, key, lastUsed }) => {
+      allIds.push({ variant, key });
+      if (!byId[variant]) {
+        byId[variant] = {};
+      }
+      byId[variant][key] = { lastUsed };
+    });
+    return {
+      expiry: state.expiry,
+      allIds,
+      byId
+    };
+  };
+
+  export const convertBiasToSearch = (state: Store.State) => {
+    // const byId = Selectors.realTimeBiasesById(state);
```

Comment:
remove comment

Reply:
That was left in since we might use the actual data on the biases at some point (if we ever decide to do decay for example), so I figured I'd leave that in but commented out. Right now all that we care is that the bias exists, so no need to have it for now.